README.md

```diff
@@ -1,6 +1,6 @@
-# Rich CLI
+# Rich-CLI
```

Comment:
[while I was there] I noticed that we had a few inconsistencies in the name of the project in that README - it seems that "Rich-CLI" is the most used name, so I normalised these 2 occurrences with this one 🙂